src/CourseDashboard.cpp

```diff
@@ -82,3 +79,42 @@ void CourseDashboard::updateUser(User & user)
         break;
     }
 }
+    
+void CourseDashboard::saveUsersToFile(std::string filename)
+{
+    std::ofstream file(filename, std::ofstream::out | std::ofstream::trunc);
+    if (file.is_open())
+    {
+        for (auto & user : users_)
+        {
+            file    << user.getName() << ';'
+                    << user.getNick() << ';'
+                    << user.getPassword() << ';'
+                    << user.getMail() << ';'
+                    << user.getGroup() << ';'
+                    << user.getGitHub() << ';'
+                    << user.getFirecode() << ';'
+                    << '\n';
```

Comment:
the User class should have a something like `serialize` method, which returns all data with proper separator. Here you could only call `file << user.serialize();`